docs/node/CATEGORIES/01-blockchain.md

```diff
@@ -17,138 +17,44 @@ const something = await client.blockchain.method()
-  blocks: number
-  headers: number
-  bestblockhash: string
-  difficulty: number
-  mediantime: number
-  verificationprogress: number
-  initialblockdownload: boolean
-  chainwork: string
-  size_on_disk: number
-  pruned: boolean
-  softforks: {
-    [id: string]: {
-      type: 'buried' | 'bip9'
-      active: boolean
-      height: number
-    }
-  }
-  warnings: string
-}
+i('https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/src/category/blockchain.ts#L233-L253')
 ```
 
 ## getBlock
 
 Get block data with a provided block header hash.
 
-```ts title="client.blockchain.getBlock()"
+```ts reference title="client.blockchain.getBlock()"
 interface blockchain {
   getBlock (hash: string, verbosity: 0): Promise<string>
   getBlock (hash: string, verbosity: 1): Promise<Block<string>>
   getBlock (hash: string, verbosity: 2): Promise<Block<Transaction>>
   getBlock<T> (hash: string, verbosity: 0 | 1 | 2): Promise<string | Block<T>>
 }
 
-interface Block<T> {
-  hash: string
-  confirmations: number
-  strippedsize: number
-  size: number
-  weight: number
-  height: number
-  masternode: string
-  minter: string
-  mintedBlocks: number
-  stakeModifier: string
-  version: number
-  versionHex: string
-  merkleroot: string
-  time: number
-  mediantime: number
-  bits: string
-  difficulty: number
-  chainwork: string
-  tx: T[]
-  nTx: number
-  previousblockhash: string
-  nextblockhash: string
-}
-
-interface Transaction {
-  txid: string
-  hash: string
-  version: number
-  size: number
-  vsize: number
-  weight: number
-  locktime: number
-  vin: Vin[]
-  vout: Vout[]
-  hex: string
-}
-
-interface Vin {
-  coinbase?: string
-  txid: string
-  vout: number
-  scriptSig: {
-    asm: string
-    hex: string
-  }
-  txinwitness?: string[]
-  sequence: string
-}
+i('https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/src/category/blockchain.ts#L255-L278')
 
-interface Vout {
-  value: BigNumber
-  n: number
-  scriptPubKey: ScriptPubKey
-  tokenId: string
-}
+i('https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/src/category/blockchain.ts#L297-L327')
 
-interface ScriptPubKey {
-  asm: string
-  hex: string
-  type: string
-  reqSigs: number
-  addresses: string[]
-}
+i('https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/src/category/blockchain.ts#L337-L343')
 ```
 
 ## getBlockHeader
 
 Get block header data with particular header hash.
 
-```ts title="client.blockchain.getBlockHeader()"
+```ts reference title="client.blockchain.getBlockHeader()"
 interface blockchain {
-  getBlockHeader (hash: string, verbosity: true): Promise<BlockHeader>
-  getBlockHeader (hash: string, verbosity: false): Promise<string>
+  i('https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/src/category/blockchain.ts#L89')
+  i('https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/src/category/blockchain.ts#L89')
```

Comment:
i think i'm outdated.. so we will be using `i(url:line)` to indicate the interface??
and these 2 lines are same.. i think the line should be diff

Reply:
The idea is using `i(url:line)` to directly import code (not only interface) from the file into the code block. It's still quite limiting actually so I am exploring other options.
 Regarding the 2 lines, I think i made a mistake, will change it.

Reply:
Interesting approach, hmm would it be possible to do symbol import instead? Since line items get shifted around when we edit the source code.

Reply:
The approach only looks clean but effectively made it even harder to maintain.
It still requires manual maintenance regardless and copy pasting the interface is easier than find out the correct GH link.